### PR TITLE
IGNITE-22283 Sql. Prepare to remove NUMBER and BITMASK native types

### PR DESCRIPTION
**Pull request by lowka**
https://issues.apache.org/jira/browse/IGNITE-22283

- Marks `NUMBER` and `BITMASK` types for removal (removal is planned in [IGNITE-22827](https://issues.apache.org/jira/browse/IGNITE-22827)).
- Remove most of the usages of these types and complimentary methods.
- `Tuple` / `BinaryTupleBuilder` remained untouched to removed the subsequence PRs.

---

Thank you for submitting the pull request.

To streamline the review process of the patch and ensure better code quality
we ask both an author and a reviewer to verify the following:

### The Review Checklist
- [ ] **Formal criteria:** TC status, codestyle, mandatory documentation. Also make sure to complete the following:  
\- There is a single JIRA ticket related to the pull request.  
\- The web-link to the pull request is attached to the JIRA ticket.  
\- The JIRA ticket has the Patch Available state.  
\- The description of the JIRA ticket explains WHAT was made, WHY and HOW.  
\- The pull request title is treated as the final commit message. The following pattern must be used: IGNITE-XXXX Change summary where XXXX - number of JIRA issue.
- [ ] **Design:** new code conforms with the design principles of the components it is added to.
- [ ] **Patch quality:** patch cannot be split into smaller pieces, its size must be reasonable.
- [ ] **Code quality:** code is clean and readable, necessary developer documentation is added if needed.
- [ ] **Tests code quality:** test set covers positive/negative scenarios, happy/edge cases. Tests are effective in terms of execution time and resources.

### Notes
- [Apache Ignite Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Java+Code+Style+Guide)

### CODE REVIEW

File: modules/table/src/test/java/org/apache/ignite/internal/table/type/NumericTypesSerializerTest.java [hunk 1]
```diff
@@ -290,33 +220,6 @@ public void testDecimalScaleTooLarge(int scale, String message) {
         );
     }
 
-    /**
-     * Test.
-     */
-    @ParameterizedTest
-    @MethodSource("sameDecimals")
-    public void testSameBinaryRepresentation(Pair<BigInteger, BigInteger> pair) {
```
Comment (ygerzhedovich) [inline modules/table/src/test/java/org/apache/ignite/internal/table/type/NumericTypesSerializerTest.java:298]: Why have you removed the test?

Comment (lowka) [reply modules/table/src/test/java/org/apache/ignite/internal/table/type/NumericTypesSerializerTest.java:298]: What is a type that corresponds to BigInteger now?

Comment (lowka) [reply modules/table/src/test/java/org/apache/ignite/internal/table/type/NumericTypesSerializerTest.java:298]: Fixed. I was confused by a type signature, that is not correct it is a pair of BigDecimals.

**Review by ygerzhedovich — APPROVED** [review]

File: modules/schema/src/test/java/org/apache/ignite/internal/schema/AbstractSchemaConverterTest.java [hunk 2]
```diff
@@ -67,15 +66,13 @@ public class AbstractSchemaConverterTest extends BaseIgniteAbstractTest {
...
-        tmp.put(NativeTypeSpec.BITMASK, Arrays.asList(null, fromBinString(""), fromBinString("1"),
-                fromBinString("10101010101010101010101")));
         tmp.put(NativeTypeSpec.STRING, Arrays.asList(null, "", UUID.randomUUID().toString()));
         tmp.put(NativeTypeSpec.BYTES, Arrays.asList(null, ArrayUtils.BYTE_EMPTY_ARRAY,
                 UUID.randomUUID().toString().getBytes(StandardCharsets.UTF_8)));
-        tmp.put(NativeTypeSpec.NUMBER, Arrays.asList(null, BigInteger.ONE, BigInteger.ZERO,
-                new BigInteger("10000000000000000000000000000000000000")));
 
         var missedTypes = new HashSet<>(Arrays.asList(NativeTypeSpec.values()));
+        missedTypes.remove(NativeTypeSpec.NUMBER);
```
Comment (xtern) [inline modules/schema/src/test/java/org/apache/ignite/internal/schema/AbstractSchemaConverterTest.java:74]: :thinking:  from my point of view all similar changes in tests must be covered with TODO "issue to remove native types"

File: modules/schema/src/test/java/org/apache/ignite/internal/schema/catalog/CatalogToSchemaDescriptorConverterTest.java [hunk 3]
```diff
@@ -58,8 +56,13 @@ public class CatalogToSchemaDescriptorConverterTest extends AbstractSchemaConver
...
     private static final int TEST_SCALE = 5;
 
+    private static Stream<NativeTypeSpec> nativeTypeSpecs() {
+        return Stream.of(NativeTypeSpec.values())
+                .filter(t -> t != NativeTypeSpec.NUMBER && t != NativeTypeSpec.BITMASK);
+    }
+
     @ParameterizedTest
-    @EnumSource(NativeTypeSpec.class)
+    @MethodSource("nativeTypeSpecs")
```
Comment (xtern) [inline modules/schema/src/test/java/org/apache/ignite/internal/schema/catalog/CatalogToSchemaDescriptorConverterTest.java:65]: from my point of view all similar changes in tests must be covered with TODO "issue to remove native types"

File: modules/schema/src/test/java/org/apache/ignite/internal/schema/marshaller/RecordMarshallerTest.java [hunk 4]
```diff
@@ -219,24 +218,7 @@ public void classWithWrongFieldType(MarshallerFactory factory) {
...
-    public void classWithIncorrectBitmaskSize(MarshallerFactory factory) {
-        SchemaDescriptor schema = new SchemaDescriptor(
-                schemaVersion.incrementAndGet(),
-                new Column[]{ new Column("key".toUpperCase(), INT32, false) },
-                new Column[]{ new Column("bitmaskCol".toUpperCase(), NativeTypes.bitmaskOf(9), true) }
-        );
-
-        RecordMarshaller<TestBitmaskObject> marshaller = factory.create(schema, TestBitmaskObject.class);
-
-        TestBitmaskObject rec = new TestBitmaskObject(1, IgniteTestUtils.randomBitSet(rnd, 42));
```
Comment (xtern) [inline modules/schema/src/test/java/org/apache/ignite/internal/schema/marshaller/RecordMarshallerTest.java:236]: `IgniteUtils.randomBitSet` is not used anymore, so from my point of view it should be removed

File: modules/schema/src/testFixtures/java/org/apache/ignite/internal/schema/SchemaTestUtils.java [hunk 5]
```diff
@@ -207,8 +190,10 @@ public static NativeType specToType(NativeTypeSpec spec) {
     public static void ensureAllTypesChecked(Stream<Column> allColumns) {
         Set<NativeTypeSpec> testedTypes = allColumns.map(c -> c.type().spec())
                 .collect(Collectors.toSet());
+        Set<NativeTypeSpec> unsupported = Set.of(NativeTypeSpec.BITMASK, NativeTypeSpec.NUMBER);
```
Comment (xtern) [inline modules/schema/src/testFixtures/java/org/apache/ignite/internal/schema/SchemaTestUtils.java:193]: this also should be marked with TODO :thinking:

File: modules/sql-engine/src/integrationTest/java/org/apache/ignite/internal/sql/engine/ItAlterTableDdlTest.java [hunk 6]
```diff
@@ -152,11 +150,6 @@ public void testNullableColumn() {
     public void testDropAndAddColumnsAllTypes() {
         List<NativeType> allTypes = SchemaTestUtils.ALL_TYPES;
 
-        Set<NativeTypeSpec> unsupportedTypes = Set.of(
-                // TODO https://issues.apache.org/jira/browse/IGNITE-18431
-                NativeTypeSpec.BITMASK
```
Comment (xtern) [inline modules/sql-engine/src/integrationTest/java/org/apache/ignite/internal/sql/engine/ItAlterTableDdlTest.java:157]: what about other places where [IGNITE-18431](https://issues.apache.org/jira/browse/IGNITE-18431) is mentioned?
Should we mention that all such places should be cleared in the `follower` ticket?

File: modules/sql-engine/src/test/java/org/apache/ignite/internal/sql/engine/util/ProjectedTupleTest.java [hunk 7]
```diff
@@ -84,6 +85,9 @@ void allTypesAreCovered() {
 
         EnumSet<NativeTypeSpec> allTypes = EnumSet.allOf(NativeTypeSpec.class);
 
+        Set<NativeTypeSpec> unsupported = Set.of(NativeTypeSpec.BITMASK, NativeTypeSpec.NUMBER);
```
Comment (xtern) [inline modules/sql-engine/src/test/java/org/apache/ignite/internal/sql/engine/util/ProjectedTupleTest.java:88]: this also better to be marked with TODO :sunglasses:

File: modules/storage-page-memory/src/test/java/org/apache/ignite/internal/storage/pagememory/index/InlineUtilsTest.java [hunk 8]
```diff
@@ -53,6 +54,7 @@ public class InlineUtilsTest extends BaseIgniteAbstractTest {
     @Test
     void testInlineSizeForNativeType() {
         EnumSet<NativeTypeSpec> nativeTypeSpecs = EnumSet.allOf(NativeTypeSpec.class);
+        nativeTypeSpecs.removeAll(Set.of(NativeTypeSpec.BITMASK, NativeTypeSpec.NUMBER));
```
Comment (xtern) [inline modules/storage-page-memory/src/test/java/org/apache/ignite/internal/storage/pagememory/index/InlineUtilsTest.java:57]: and this

File: modules/system-view/src/test/java/org/apache/ignite/internal/systemview/SystemViewManagerTest.java [hunk 9]
```diff
@@ -163,6 +164,11 @@ public void registerAllColumnTypes(NativeTypeSpec typeSpec) {
         assertTrue(viewMgr.completeRegistration().isDone());
     }
 
+    private static Stream<NativeTypeSpec> nativeTypeSpecs() {
```
Comment (xtern) [inline modules/system-view/src/test/java/org/apache/ignite/internal/systemview/SystemViewManagerTest.java:167]: and this

File: modules/table/src/integrationTest/java/org/apache/ignite/internal/table/ItColocationTest.java [hunk 10]
```diff
@@ -373,9 +367,13 @@ private static Object generateValueByType(int i, NativeTypeSpec type) {
 
     private static Stream<Arguments> twoColumnsParameters() {
         List<Arguments> args = new ArrayList<>();
+        Set<NativeTypeSpec> unsupported = Set.of(NativeTypeSpec.BITMASK, NativeTypeSpec.NUMBER);
 
         for (NativeTypeSpec t0 : NativeTypeSpec.values()) {
             for (NativeTypeSpec t1 : NativeTypeSpec.values()) {
+                if (unsupported.contains(t0) || unsupported.contains(t1)) {
```
Comment (xtern) [inline modules/table/src/integrationTest/java/org/apache/ignite/internal/table/ItColocationTest.java:374]: and this

File: modules/table/src/test/java/org/apache/ignite/internal/table/KeyValueViewOperationsSimpleSchemaTest.java [hunk 11]
```diff
@@ -490,7 +489,12 @@ public void putGetAllTypes() {
         );
 
         // Validate all types are tested.
-        assertEquals(Set.of(NativeTypeSpec.values()),
+        Set<NativeTypeSpec> nativeTypes = EnumSet.allOf(NativeTypeSpec.class);
+
+        Set<NativeTypeSpec> unsupported = Set.of(NativeTypeSpec.BITMASK, NativeTypeSpec.NUMBER);
```
Comment (xtern) [inline modules/table/src/test/java/org/apache/ignite/internal/table/KeyValueViewOperationsSimpleSchemaTest.java:494]: and this

File: modules/table/src/test/java/org/apache/ignite/internal/table/KeyValueViewOperationsTest.java [hunk 12]
```diff
@@ -668,7 +666,11 @@ private KeyValueViewImpl<TestKeyObject, TestObjectWithAllTypes> kvView() {
         // Validate all types are tested.
         Set<NativeTypeSpec> testedTypes = Arrays.stream(valCols).map(c -> c.type().spec())
                 .collect(Collectors.toSet());
+
+        Set<NativeTypeSpec> unsupported = Set.of(NativeTypeSpec.BITMASK, NativeTypeSpec.NUMBER);
+
         Set<NativeTypeSpec> missedTypes = Arrays.stream(NativeTypeSpec.values())
+                .filter(t -> !unsupported.contains(t))
```
Comment (xtern) [inline modules/table/src/test/java/org/apache/ignite/internal/table/KeyValueViewOperationsTest.java:673]: and this

File: modules/table/src/test/java/org/apache/ignite/internal/table/MutableRowTupleAdapterTest.java [hunk 13]
```diff
@@ -470,11 +468,17 @@ public void testFullSchemaHasAllTypes() {
...
                 .collect(Collectors.toSet());
 
+        Set<ColumnType> unsupported = Set.of(ColumnType.BITMASK, ColumnType.NUMBER);
+
         for (ColumnType columnType : ColumnType.values()) {
             if (columnType == ColumnType.NULL) {
                 continue;
             }
 
+            if (unsupported.contains(columnType)) {
```
Comment (xtern) [inline modules/table/src/test/java/org/apache/ignite/internal/table/MutableRowTupleAdapterTest.java:478]: and this

File: modules/table/src/test/java/org/apache/ignite/internal/table/RecordViewOperationsTest.java [hunk 14]
```diff
@@ -435,7 +433,11 @@ private RecordViewImpl<TestObjectWithAllTypes> recordView() {
         // Validate all types are tested.
         Set<NativeTypeSpec> testedTypes = Arrays.stream(valCols).map(c -> c.type().spec())
                 .collect(Collectors.toSet());
+
+        Set<NativeTypeSpec> unsupported = Set.of(NativeTypeSpec.BITMASK, NativeTypeSpec.NUMBER);
+
         Set<NativeTypeSpec> missedTypes = Arrays.stream(NativeTypeSpec.values())
+                .filter(t -> !unsupported.contains(t))
```
Comment (xtern) [inline modules/table/src/test/java/org/apache/ignite/internal/table/RecordViewOperationsTest.java:440]: and this

**Pull request merged**